test/hotspot/jtreg/runtime/Safepoint/TestAbortVMOnSafepointTimeout.java

```diff
@@ -78,4 +112,21 @@ public static void main(String[] args) throws Exception {
             }
         }
     }
+
+    public static class TestWithDelay {
+
+        public static final String PRE_STALL_TEXT = "THE FOLLOWING STALL SHOULD BE CAPTURED";
+
+        public static void main(String[] args) throws Exception {
+            WhiteBox wb = WhiteBox.getWhiteBox();
+            // induce a stall that should not be picked up before grace period
+            wb.waitUnsafe(999);
```

Comment:
So if we reach this line after 500ms from the time we set the initial counter in os::init() we will trigger the abort here instead. I measured that time just to check how close we are and in my testing machine I see we get here at about 130ms so probably is good enough, but maybe we should add 1 second to AbortVMOnSafepointTimeoutDelay just to be safe?